Controller/ProcessReportController.php

```diff
@@ -0,0 +1,126 @@
+     */
+    public function index(
+        Request $request,
+        Process $process,
+        TranslatorInterface $translator,
+        EconomyEntryRepository $economyEntryRepository,
+        ConclusionService $conclusionService,
+        JournalEntryRepository $journalEntryRepository
+    ): Response {
+        $form = $this->createFormBuilder()
+            ->add('options', ChoiceType::class, [
+                'label' => $translator->trans('process_report.form.choices.placeholder'),
+                'choices' => [
+                    $translator->trans('process_report.form.choices.internal_notes') => 'internal_notes',
+                    $translator->trans('process_report.form.choices.only_summary') => 'only_summary',
+                ],
+            ])
+            ->add('generate', SubmitType::class, [
+                'label' => $translator->trans('process_report.form.submit'),
+            ])
+            ->getForm();
+
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            $formData = $form->getData();
+
+            $viewData = [
+                'process' => $process,
+                'choice' => $formData['options'],
+            ];
+
+            $viewData['economyEntries'] = [
+                'service' => $economyEntryRepository->findBy([
+                    'process' => $process,
+                    'type' => EconomyEntryEnumType::SERVICE,
+                ]),
+                'account' => $economyEntryRepository->findBy([
+                   'process' => $process,
+                   'type' => EconomyEntryEnumType::ACCOUNT,
+                ]),
+                'arrear' => $economyEntryRepository->findBy([
+                   'process' => $process,
+                   'type' => EconomyEntryEnumType::ARREAR,
+                ]),
+            ];
+
+            $qb = $journalEntryRepository->createQueryBuilder('journalEntry');
+            $qb
+                ->select('journalEntry')
+                ->where('journalEntry.process = :process')
+                ->setParameter('process', $process)
+                ->andWhere('journalEntry.type = :note')
+                ->setParameter('note', JournalEntryEnumType::NOTE);
+
+            if ('internal_notes' === $formData['options']) {
+                $qb
+                    ->orWhere('journalEntry.type = :internal')
+                    ->setParameter('internal', JournalEntryEnumType::INTERNAL_NOTE);
+            }
+
+            $viewData['journalEntries'] = $qb->getQuery()->getArrayResult();
+
+            $viewData['conclusionTemplate'] = $conclusionService->getTemplate(
+                \get_class($process->getConclusion()),
+                '',
+                '@KontrolgruppenCore/process_report/'
+            );
+
+            $report = $this->renderView('@KontrolgruppenCore/process_report/_report.html.twig', $viewData);
+
+            // @TODO Should probably be in a Service
```

Comment:
Probably all of the report code should be moved to a service, with the form result as parameter.

Reply:
Fixed in 7424d35